server/apps/immich/src/api-v1/device-info/device-info.controller.ts

```diff
@@ -14,19 +13,23 @@ import { DeviceInfoResponseDto } from './response-dto/create-device-info-respons
 export class DeviceInfoController {
   constructor(private readonly deviceInfoService: DeviceInfoService) {}
 
+  /** @deprecated */
   @Post()
-  async createDeviceInfo(
-    @Body(ValidationPipe) createDeviceInfoDto: CreateDeviceInfoDto,
-    @GetAuthUser() authUser: AuthUserDto,
-  ): Promise<DeviceInfoResponseDto> {
-    return this.deviceInfoService.create(createDeviceInfoDto, authUser);
+  public async createDeviceInfo(@GetAuthUser() user: AuthUserDto, @Body(ValidationPipe) dto: UpsertDeviceInfoDto) {
+    const deviceInfo = await this.deviceInfoService.upsert({ ...dto, userId: user.id });
+    return mapDeviceInfoResponse(deviceInfo);
   }
 
+  /** @deprecated */
   @Patch()
-  async updateDeviceInfo(
-    @Body(ValidationPipe) updateDeviceInfoDto: UpdateDeviceInfoDto,
-    @GetAuthUser() authUser: AuthUserDto,
-  ): Promise<DeviceInfoResponseDto> {
-    return this.deviceInfoService.update(authUser.id, updateDeviceInfoDto);
+  public async updateDeviceInfo(@GetAuthUser() user: AuthUserDto, @Body(ValidationPipe) dto: UpsertDeviceInfoDto) {
+    const deviceInfo = await this.deviceInfoService.upsert({ ...dto, userId: user.id });
+    return mapDeviceInfoResponse(deviceInfo);
+  }
+
+  @Put()
+  public async upsertDeviceInfo(@GetAuthUser() user: AuthUserDto, @Body(ValidationPipe) dto: UpsertDeviceInfoDto) {
```

Comment:
this function signature does not have a return type. yet it returns deviceInfo

Reply:
Added the response dto to all three methods.